From 250dc372cfb64dfb21db2e7bd67ed28822aa132d Mon Sep 17 00:00:00 2001
From: Phil Pennock <pdp@exim.org>
Date: Wed, 5 Jun 2019 05:35:28 -0400
Subject: [PATCH 27/34] Unbreak heimdal_gssapi auth driver

Commit 251b9eb46 broke heimdal_gssapi by changing the function
definition in the `.c` without changing the declaration in the `.h`.
Was part of 4.92.

Make corresponding `.h` change to reflect newer internal API.

(cherry picked from commit 6ee110613402e8562c03b4a11c3ffbdbd47bd153)
(cherry picked from commit 40fe3ea73eb7524a6143755854633ed8392d39b4)
(cherry picked from commit 171adf11d72efb4781a3028a849e0ed6e521a4fa)
---
 doc/ChangeLog          | 2 ++
 src/auths/heimdal_gssapi.h | 3 +--
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/doc/ChangeLog b/doc/ChangeLog
index a3a6b9b39..67dfafa7a 100644
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -80,6 +80,8 @@ JH/27 Bug 2404: Use the main-section configuration option "dsn_from" for
       success-DSN messages.  Previously the From: header was always the default
       one for these; the option was ignored.
 
+PP/01 Unbreak heimdal_gssapi, broken in 4.92.
+
 
 Exim version 4.92
 -----------------
diff --git a/src/auths/heimdal_gssapi.h b/src/auths/heimdal_gssapi.h
index a606a5c26..b682b5ff6 100644
--- a/src/auths/heimdal_gssapi.h
+++ b/src/auths/heimdal_gssapi.h
@@ -32,8 +32,7 @@ extern auth_heimdal_gssapi_options_block auth_heimdal_gssapi_option_defaults;
 
 extern void auth_heimdal_gssapi_init(auth_instance *);
 extern int auth_heimdal_gssapi_server(auth_instance *, uschar *);
-extern int auth_heimdal_gssapi_client(auth_instance *, smtp_inblock *,
-                                smtp_outblock *, int, uschar *, int);
+extern int auth_heimdal_gssapi_client(auth_instance *, void *, int, uschar *, int);
 extern void auth_heimdal_gssapi_version_report(FILE *f);
 
 /* End of heimdal_gssapi.h */
-- 
2.23.0.rc1

